zerver/views/streams.py

```diff
@@ -629,6 +630,10 @@ def add_subscriptions_backend(
     # Newly created streams are also authorized for the creator
     streams = authorized_streams + created_streams
 
+    if is_default_stream:
+        for stream in created_streams:
+            do_add_default_stream(stream)
```

Comment:
We might want to change this in future to use `bulk_create` rather than using a loop.

Reply:
Please! update me if we want this change.

Reply:
Yes, this refactor would be a good idea. I think it's OK to do it in a separate prep or follow-up PR.